Microsoft.SystemForCrossDomainIdentityManagement/Protocol/ProtocolExtensions.cs

```diff
@@ -148,9 +148,14 @@ private static void Apply(this Core2Group group, PatchOperation2 operation)
                                                  Value = item.Value
                                              })
                                      .ToArray();
-                                group.Members =
-                                    group.Members != null ?
-                                        group.Members.Concat(newMembers).ToArray() : newMembers;
+
+                                foreach(Member member in newMembers)
+                                {
+                                    if (!group.Members.Any(M => M.Value == member.Value))
+                                    {
+                                        group.Members = group.Members.Concat(new [] {member});
```

Comment:
this is highly non-performant. For every member that needs to be added to the group, this code scans all the other members, so the time complexity is O(n). This shall scale really poorly for the groups of a decent size.